frame/democracy/src/vote.rs

```diff
@@ -51,8 +52,21 @@ impl Decode for Vote {
 	}
 }
 
+impl TypeInfo for Vote {
+	type Identity = Self;
+
+	fn type_info() -> scale_info::Type {
+		scale_info::Type::builder()
+			.path(scale_info::Path::new("Vote", module_path!()))
+			.composite(
+				scale_info::build::Fields::unnamed()
+					.field(|f| f.ty::<u8>().docs(&["Raw vote byte, encodes aye + conviction"])),
+			)
+	}
+}
```

Comment:
Because of the custom encoding

`output.push_byte(u8::from(self.conviction) | if self.aye { 0b1000_0000 } else { 0 });`

This can't be described in `TypeInfo`, so the approach I have taken is to describe the opaque encoding, in order that we can still decode the raw value without any additional logic supplied by a client.

Obviously downstream consumers will have to provide further custom decoding if they want to actually extract the fields, but that would be true anyway.

Reply:
Ahh I missed that. Makes sense now.